ml-dhs-aws/src/test/java/com/marklogic/client/dhs/test/TestJobRunner.java

```diff
@@ -37,11 +62,138 @@ public void testRun() throws IOException {
+
+		JobRunner jobRunner = new JobRunner();
+		String[] roles = { "admin" };
+		jobRunner.setRoles(roles);
+		JSONDocumentManager jsonMgr = client.newJSONDocumentManager();
+
+		try (InputStream csvStream = openStream(csvFile);
+				InputStream jobStream = openStream(JobRunner.jobFileFor(csvFile));) {
+			jobRunner.run(client, csvStream, jobStream);
+
+			ObjectMapper objectMapper = new ObjectMapper();
+
+			ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobStream);
+			ObjectNode jobDef = (ObjectNode) jobControlObj.get("job");
+
+			String beforeDocId = JobRunner.getBeforeJobDocumentUri(jobDef.path("id").asText());
+			assertNotNull(jsonMgr.exists(beforeDocId));
+
+			DocumentMetadataHandle beforeDocumentMetadata = new DocumentMetadataHandle();
+			JacksonHandle readHandle = new JacksonHandle();
+			
+			assertNotNull(jsonMgr.read(beforeDocId, beforeDocumentMetadata, readHandle)); 
+			assertTrue(beforeDocumentMetadata.getCollections().contains(JobRunner.getJobCollection(jobDef.path("id").asText())));
+			assertTrue(beforeDocumentMetadata.getCollections().contains("/beforeJob"));
+			assertTrue(beforeDocumentMetadata.getPermissions().containsKey("admin"));
+			assertTrue(beforeDocumentMetadata.getPermissions().get("admin").contains(Capability.READ));
+			assertTrue(beforeDocumentMetadata.getPermissions().get("admin").contains(Capability.UPDATE));
+
+		} catch(Exception ex) {
+			ex.printStackTrace();
+		}
+	}
+   
+	@Test
+	public void testAfterJobDocument() throws Exception {
+		String csvFile = "data"+File.separator+"OrderLines.csv";
+
+		JobRunner jobRunner = new JobRunner();
+		String[] roles = { "admin" };
+		jobRunner.setRoles(roles);
+		JSONDocumentManager jsonMgr = client.newJSONDocumentManager();
+
+		try (InputStream csvStream = openStream(csvFile);
+				InputStream jobStream = openStream(JobRunner.jobFileFor(csvFile));) {
+			jobRunner.run(client, csvStream, jobStream);
+
+			ObjectMapper objectMapper = new ObjectMapper();
+
+			ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobStream);
+			ObjectNode jobDef = (ObjectNode) jobControlObj.get("job");
+
+			String afterDocId = JobRunner.getAfterJobDocumentUri(jobDef.path("id").asText());
+			assertNotNull(jsonMgr.exists(afterDocId));
+			DocumentMetadataHandle afterDocumentMetadata = new DocumentMetadataHandle();
+			JacksonHandle readHandle = new JacksonHandle();
+			
+			assertNotNull(jsonMgr.read(afterDocId, afterDocumentMetadata, readHandle)); 
+			assertTrue(afterDocumentMetadata.getCollections().contains(JobRunner.getJobCollection(jobDef.path("id").asText())));
+			assertTrue(afterDocumentMetadata.getCollections().contains("/afterJob"));
+			assertTrue(afterDocumentMetadata.getPermissions().containsKey("admin"));
+			assertTrue(afterDocumentMetadata.getPermissions().get("admin").contains(Capability.READ));
+			assertTrue(afterDocumentMetadata.getPermissions().get("admin").contains(Capability.UPDATE));
+
```

Comment:
As with the beforeDocument, could their be assertions on the populated readHandle for the afterDocument?